pennylane/ftqc/decomposition.py

```diff
@@ -43,3 +54,303 @@ def convert_to_mbqc_gateset(tape):
         )
     tapes, fn = decompose(tape, gate_set=mbqc_gate_set, alt_decomps={Rot: [_rot_to_xzx]})
     return tapes, fn
+
+
+@transform
+def convert_to_mbqc_formalism(tape):
+    """Convert a circuit to the textbook MBQC formalism based on the procedures outlined in
+    Raussendorf et al. 2003, https://doi.org/10.1103/PhysRevA.68.022312. The circuit must
+    be decomposed to the gate set {CNOT, H, S, RotXZX, RZ, X, Y, Z, Identity, GlobalPhase}
+    before applying the transform.
+
+    Note that this transform leaves all Paulis and Identities as physical gates, and applies
+    all byproduct operations online immediately after their respective measurement procedures."""
+
+    if len(tape.measurements) != 1 or not isinstance(tape.measurements[0], (SampleMP)):
+        raise NotImplementedError(
+            "Transforming to the MBQC formalism is not implemented for circuits where the "
+            "final measurements have not been converted to a single samples measurement"
+        )
+
+    mp = tape.measurements[0]
+    meas_wires = mp.wires if mp.wires else tape.wires
+
+    # we include 13 auxillary wires - the largest number needed is 13 (for CNOT)
+    num_qubits = len(tape.wires) + 13
+    q_mgr = QubitMgr(num_qubits=num_qubits, start_idx=0)
+
+    wire_map = {w: q_mgr.acquire_qubit() for w in tape.wires}
+
+    with AnnotatedQueue() as q:
+        for op in tape.operations:
+            if isinstance(op, GlobalPhase):  # no wires
+                GlobalPhase(*op.data)
+            elif isinstance(op, CNOT):  # two wires
+                ctrl, tgt = op.wires[0], op.wires[1]
+                wire_map[ctrl], wire_map[tgt], measurements = queue_cnot(
+                    q_mgr, wire_map[ctrl], wire_map[tgt]
+                )
+                cnot_corrections(measurements)(wire_map[ctrl], wire_map[tgt])
+            else:  # one wire
+                if isinstance(op, X | Y | Z | Identity):
```

Comment:
```suggestion
                if isinstance(op, PauliX | PauliY | PauliZ | Identity):
```

Reply:
Unfortunately using PauliX etc did not seem to make it happier 😞